src/Pages/Home.jsx

```diff
@@ -1,11 +1,25 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import Cart from "../Components/Cart";
 import NavBar from "../Components/NavBar";
 import ProductTable from "../Components/ProductTable";
 
 function Home() {
   const [open, setOpen] = useState(false);
-  const [cart, updateCart] = useState([]);
+  const [cart, updateCart] = useState(() => {
+    const localCart = JSON.parse(localStorage.getItem('cart'))
+    return localCart ? localCart : []
+  });
+
+  // Trying to have one location to update the local storage when the cart state changes, but only when items are added or removed.
+  // It doesn't make sense to overwrite the localstorage every time the page is loaded with the logic I have with state, above.
+  // But this implementation isn't much better, since I'm reading from disk twice on first load, and then again every time the cart is updated.
+  // The alternative that I'm aware of is to put it in the onClick actions for adding and removing from the cart, which probably makes more sense at this point in time.
```

Comment:
You're on the right track - linking the actions that update the cart to localStorage is what I would recommend, but with a slight twist.

instead of spreading the code to multiple components (as you stated), you can simply wrap the updateCart function in a new function. This new function can update the cart state, and also pass the new cart state straight into localStorage. [Here is an example](https://github.com/developer-job-simulation/react-ecommerce/pull/59/commits/77ad98a982970e061b6c47f9cce449bdb54adfa0#diff-2556b2256aa32b258d4c2716b13ba791d50399bf54b747e98108b6948f00a10bR8).

Reply:
Ohhh I think I see what you did there. That makes sense, I didn't think of handling state within another function. Thank you!